src/Workspaces/Core/Portable/Workspace/ProjectSystem/IHostDiagnosticAnalyzerProvider.cs

```diff
@@ -14,4 +14,6 @@ namespace Microsoft.CodeAnalysis.Workspaces.ProjectSystem;
 internal interface IHostDiagnosticAnalyzerProvider
 {
     ImmutableArray<(AnalyzerFileReference reference, string extensionId)> GetAnalyzerReferencesInExtensions();
+
+    ImmutableArray<(string path, string extensionId)> GetRazorReferencesInExtensions();
```

Comment:
Consider adding documentation to this method or clarifying its name - it is supposed to return only razor *source generators* not *analyzers*, right?

Reply:
Well, it returns anything that's marked as `Microsoft.VisualStudio.RazorAssembly`. Those can still have analyzers or generators (or nothing) in them. I'll change the name to `RazorAssemblies` to make it clearer.